subprojects/core/src/test/groovy/org/gradle/api/internal/tasks/options/OptionReaderTest.groovy

```diff
@@ -274,53 +302,91 @@ class OptionReaderTest extends Specification {
 
         options[5].name == "field5"
         options[5].description == "Descr Field5"
-        options[5].argumentType == List
+        options[5].argumentType == String
         options[5].availableValues.isEmpty()
+
+        options[6].name == "field6"
+        options[6].description == "Descr Field6"
+        options[6].argumentType == List
+        options[6].availableValues.isEmpty()
+
+        options[7].name == "field7"
+        options[7].description == "Descr Field7"
+        options[7].argumentType == List
+        options[7].availableValues.isEmpty()
+
+        options[8].name == "field8"
+        options[8].description == "Descr Field8"
+        options[8].argumentType == List
+        options[8].availableValues.isEmpty()
```

Comment:
❓ This index based system is becoming really hard to read, and I guess it's also error-prone. Would it be a lot of work to switch to a key-based system such that we could do things like `options["field8"].availableValues.isEmpty()`?

Reply:
I'm guessing we are not going to change the indexed system here?